src/doc/src/reference/semver.md

```diff
@@ -204,6 +206,767 @@ This is not considered a major change because conventionally glob imports are
 a known forwards-compatibility hazard. Glob imports of items from external
 crates should be avoided.
 
+### Major: Changing the alignment, layout, or size of a well-defined type {#type-layout}
+
+It is a breaking change to change the alignment, layout, or size of a type that was previously well-defined.
+
+In general, types that use the [the default representation] do not have a well-defined alignment, layout, or size.
+The compiler is free to alter the alignment, layout or size, so code should not make any assumptions about it.
+
+> **Note**: It may be possible for external crates to break if they make assumptions about the alignment, layout, or size of a type even if it is not well-defined.
+> This is not considered a SemVer breaking change since those assumptions should not be made.
+
+Some examples of changes that are not a breaking change are (assuming no other rules in this guide are violated):
+
+* Adding, removing, or changing fields of a default representation struct, union, or enum in such a way that the change follows the other rules in this guide (for example, using `non_exhaustive` to allow those changes, or changes to private fields that are already private).
```

Comment:
```suggestion
* Adding, removing, reordering, or changing fields of a default representation struct, union, or enum in such a way that the change follows the other rules in this guide (for example, using `non_exhaustive` to allow those changes, or changes to private fields that are already private).
```

Reply:
This is an interesting one: is changing enum variant order of a default-repr enum a breaking change?

Enum variant order _can_ be observed in several ways, even for default-repr enums. This issue lists a few of them: https://github.com/obi1kenobi/cargo-semver-checks/issues/305#issuecomment-1479860160

If we have an official determination of whether this is considered a major change or not, I will make `cargo-semver-checks` check for it accordingly.

Reply:
Oh good point. For tuples it would also be breaking due to the automatic field naming.

But still, reordering should be mentioned because that's definitely a change people do make and that may or may not get reflected in the layout.